frontend/src/lib/notify/index.ts

```diff
@@ -30,6 +30,10 @@ export class NotificationService {
     this.addNotification({ message, duration, category: 'alert-warning' });
   }
 
+  notifyPersistent = (message: string, category?: 'alert-warning'): void => {
+    this.addPersistentNotification({ message, duration: Infinity, category });
+  }
+
```

Comment:
I'd rather handle this by tweaking the existing notifySuccess and notifyWarning functions to simply treat a duration of 0 (or negative) as being infinite, and not add a new `notifyPersistent` function. See my comment below on `addNotification`.